swap/simulations_test.go

```diff
@@ -676,33 +682,60 @@ func TestBasicSwapSimulation(t *testing.T) {
 	log.Info("Simulation ended")
 }
 
-func waitForChequeProcessed(t *testing.T, ts *testService, counter metrics.Counter, lastCount int64) error {
+func waitForChequeProcessed(t *testing.T, backend *swapTestBackend, counter metrics.Counter, lastCount int64, p *Peer, expectedLastPayout uint64) error {
 	t.Helper()
 	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
 	defer cancel()
 
 	// we are going to wait for two things:
-	// * that a cheque has been processed
+	// * that a cheque has been processed and confirmed
 	// * that the other side actually did process the testMsgPrice message
 	//   (by checking that the message counter has been increased)
 	var wg sync.WaitGroup
-	wg.Add(2)
+	var err error
+	wg.Add(3)
 
 	go func() {
-		backend := ts.swap.backend.(*swapTestBackend)
-		select {
-		case <-ctx.Done():
-			t.Fatal("timed out waiting for cheque to be processed")
-		case <-backend.cashDone:
-			wg.Done()
+		for {
+			select {
+			case <-ctx.Done():
+				err = errors.New("Timed out waiting for cheque to have been cashed")
```

Comment:
I wonder what happens if multiple of those waiting functions time out. `err` gets written to multiple times and we get the error of the last go routine to finish? I wonder if it would make sense to log the errors here, so in case the test fails we see all the things that went wrong at once instead of just the "last" error.

Reply:
Yes, good point, because really if it fails it doesn't matter that much, but to be able to fix the issue we should know the one which failed.